client/http/client.go

```diff
@@ -0,0 +1,280 @@
+func WithTLSConfig(tlsConf *tls.Config) ClientOption {
+	return func(c *client) {
+		c.tlsConf = tlsConf
+	}
+}
+
+// NewClient creates a PD HTTP client with the given PD addresses and TLS config.
+func NewClient(
+	pdAddrs []string,
+	opts ...ClientOption,
+) Client {
+	c := &client{}
+	// Apply the options first.
+	for _, opt := range opts {
+		opt(c)
+	}
+	// Normalize the addresses with correct scheme prefix.
+	for i, addr := range pdAddrs {
+		if !strings.HasPrefix(addr, "http") {
+			if c.tlsConf != nil {
+				addr = "https://" + addr
+			} else {
+				addr = "http://" + addr
+			}
+			pdAddrs[i] = addr
+		}
+	}
+	c.pdAddrs = pdAddrs
+	// Init the HTTP client if it's not configured.
+	if c.cli == nil {
+		c.cli = &http.Client{Timeout: defaultTimeout}
+		if c.tlsConf != nil {
+			transport := http.DefaultTransport.(*http.Transport).Clone()
+			transport.TLSClientConfig = c.tlsConf
+			c.cli.Transport = transport
+		}
+	}
+
+	return c
+}
+
+// Close closes the HTTP client.
+func (c *client) Close() {
+	if c.cli != nil {
+		c.cli.CloseIdleConnections()
+	}
+	log.Info("[pd] http client closed")
+}
+
+func (c *client) pdAddr() string {
+	// TODO: support the customized PD address selection strategy.
+	return c.pdAddrs[0]
+}
+
+func (c *client) request(
+	ctx context.Context,
+	name, uri string,
+	res interface{},
+) error {
+	reqURL := fmt.Sprintf("%s%s", c.pdAddr(), uri)
+	logFields := []zap.Field{
+		zap.String("name", name),
+		zap.String("url", reqURL),
+	}
+	log.Debug("[pd] request the http url", logFields...)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, reqURL, nil)
+	if err != nil {
+		log.Error("[pd] create http request failed", append(logFields, zap.Error(err))...)
+		return errors.Trace(err)
+	}
+	// TODO: integrate the metrics.
+	resp, err := c.cli.Do(req)
+	if err != nil {
+		log.Error("[pd] do http request failed", append(logFields, zap.Error(err))...)
+		return errors.Trace(err)
+	}
+	defer func() {
+		err = resp.Body.Close()
+		if err != nil {
+			log.Warn("[pd] close http response body failed", append(logFields, zap.Error(err))...)
+		}
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		logFields = append(logFields, zap.String("status", resp.Status))
+
+		bs, readErr := io.ReadAll(resp.Body)
+		if readErr != nil {
+			logFields = append(logFields, zap.NamedError("read-body-error", err))
+		} else {
+			logFields = append(logFields, zap.ByteString("body", bs))
+		}
+
+		log.Error("[pd] request failed with a non-200 status", logFields...)
+		return errors.Errorf("request pd http api failed with status: '%s'", resp.Status)
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(res)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// GetRegionByID gets the region info by ID.
+func (c *client) GetRegionByID(ctx context.Context, regionID uint64) (*RegionInfo, error) {
+	var region RegionInfo
+	err := c.request(ctx, "GetRegionByID", RegionByID(regionID), &region)
```

Comment:
TiDB has a metric like `tidb_server_pd_api_execution_duration`.  before using this `sdk`, It should also be considered to pass the metric into the client for compatibility.

Reply:
Done.

Reply:
Local test with replaced PD HTTP client:

![image](https://github.com/tikv/pd/assets/1446531/e1e9d328-e3bf-41df-8197-e1a98c41d923)